File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_voltage_current'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
vc = BrickletVoltageCurrent.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 10 seconds (10000ms)
vc.set_debounce_period 10000

# Register threshold reached callback for power greater than 10 W (parameter has unit mW)
vc.register_callback(BrickletVoltageCurrent::CALLBACK_POWER_REACHED) do |power|
  puts "Power: #{power/1000.0} W"
end

# Configure threshold for "greater than 10 W" (unit is mW)
vc.set_power_callback_threshold '>', 10*1000, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/php/ExampleThreshold.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletVoltageCurrent.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletVoltageCurrent;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Voltage/Current Bricklet

// Callback function for power reached callback (parameter has unit mW)
function cb_powerReached($power)
{
    echo "Power: " . $power/1000.0 . " W\n";
}

$ipcon = new IPConnection(); // Create IP connection
$vc = new BrickletVoltageCurrent(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
$vc->setDebouncePeriod(10000);

// Register power reached callback to function cb_powerReached
$vc->registerCallback(BrickletVoltageCurrent::CALLBACK_POWER_REACHED, 'cb_powerReached');

// Configure threshold for power "greater than 10 W" (unit is mW)
$vc->setPowerCallbackThreshold('>', 10*1000, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
